frontend_tests/node_tests/event_status.js

```diff
@@ -0,0 +1,345 @@
+"use strict";
+
+const {strict: assert} = require("assert");
+
+const {set_global, zrequire, mock_esm} = require("../zjsunit/namespace");
+const {run_test} = require("../zjsunit/test");
+const $ = require("../zjsunit/zjquery");
+
+const event_status = zrequire("billing/event_status");
+const helpers = mock_esm("../../static/js/billing/helpers");
+
+run_test("initialize_retry_with_another_card_link_click_handler", ({override}) => {
+    override(helpers, "create_ajax_request", (url, form_name, ignored_inputs, method, callback) => {
+        assert.equal(url, "/json/billing/session/start_retry_payment_intent_session");
+        assert.equal(form_name, "restartsession");
+        assert.deepEqual(ignored_inputs, []);
+        assert.equal(method, "POST");
+        set_global("window", {
+            location: {
+                replace: (new_location) => {
+                    assert.equal(new_location, "stripe_session_url");
+                },
+            },
+        });
+        callback({stripe_session_url: "stripe_session_url"});
+    });
+    event_status.initialize_retry_with_another_card_link_click_handler();
+    const retry_click_handler = $("#retry-with-another-card-link").get_on_handler("click");
+    retry_click_handler({preventDefault: () => {}});
+});
+
+run_test("check_status", async ({override}) => {
```

Comment:
Our test runner has no real support for asynchronous tests. An `async` function doesn’t run until after all other tests have completed, the namespace restoring has already occured, etc. This is now causing tests to fail on `main`.

(I take this as further evidence that we should move to a real test runner.)